### PR TITLE
Reworking of parameters/defaults for protein FDR calculations

### PR DESCRIPTION
**Pull request by hendrikweisser**
This addresses issue #1449. The "decoy_string" and "annotate_proteins" parameters were removed from FalseDiscoveryRate and PeptideIndexer, respectively. Protein-level FDR calculations are now based on annotations from PeptideIndexer, which now always annotates proteins. Also changed FalseDiscoveryRate's "q_value" parameter into a proper flag ("false" by default) termed "no_qvalues".

### CODE REVIEW

File: src/openms/source/ANALYSIS/ID/FalseDiscoveryRate.cpp [hunk 1]
```diff
@@ -48,29 +48,28 @@ namespace OpenMS
   FalseDiscoveryRate::FalseDiscoveryRate() :
     DefaultParamHandler("FalseDiscoveryRate")
   {
-    defaults_.setValue("q_value", "true", "If 'true', the q-values will be calculated instead of the FDRs");
-    defaults_.setValidStrings("q_value", ListUtils::create<String>("true,false"));
+    defaults_.setValue("no_qvalues", "false", "If 'true' strict FDRs will be calculated instead of q-values (the default)");
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/FalseDiscoveryRate.cpp]: is this change required? because it will break existing pipelines

File: src/topp/PeptideIndexer.cpp [hunk 2]
```diff
@@ -998,15 +991,12 @@ class TOPPPeptideIndexer :
...
+    for (vector<ProteinIdentification>::iterator id_it = prot_ids.begin(); id_it != prot_ids.end(); ++id_it)
     {
-      for (vector<ProteinIdentification>::iterator id_it = prot_ids.begin(); id_it != prot_ids.end(); ++id_it)
+      for (vector<ProteinHit>::iterator hit_it = id_it->getHits().begin(); hit_it != id_it->getHits().end(); ++hit_it)
       {
-        for (vector<ProteinHit>::iterator hit_it = id_it->getHits().begin(); hit_it != id_it->getHits().end(); ++hit_it)
-        {
-          hit_it->setMetaValue("target_decoy", (protein_is_decoy[hit_it->getAccession()] ? "decoy" : "target"));
-        }
+        hit_it->setMetaValue("target_decoy", (protein_is_decoy[hit_it->getAccession()] ? "decoy" : "target"));
```
Comment (timosachsenberg) [inline src/topp/PeptideIndexer.cpp]: target+decoy is also a valid annotation. Do you know if this is handled properly here? I am also missing a few tests for "target+decoy"

Comment (timosachsenberg) [inline src/topp/PeptideIndexer.cpp]: Ah sorry. This only applies to PeptideHits

**Comment by timosachsenberg** [comment]
Nice. Please consider the need to change this to a proper flag. 
If we need to break pipelines we could think about using a stringlist "q-value" "FDR" instead of a flag instead and call the parameter e.g. "report"

**Comment by hendrikweisser** [comment]
> Nice. Please consider the need to change this to a proper flag.

Do you think this parameter is actually used by anyone? I can't really think of a case where you would NOT want q-values, and the fact that there are no tests for the "no q-values" case suggests to me that it's not a common case.

> If we need to break pipelines we could think about using a stringlist "q-value" "FDR" instead of a flag instead and call the parameter e.g. "report"

Yes, this would be a bit more explicit, but I don't think it's worth the effort for a parameter that probably nobody is using. In general I prefer a simple flag if there are only two alternatives.

**Comment by timosachsenberg** [comment]
I am just imaging that the tool FalseDiscoveryRate might be used to calculate FDRs in a different way (target decoy based is only one way to do so i.e. PEP based is also possible ). An explicit string might allow to extend it without breaking workflows again.
I got bit more cautious when working with the KNIME guys and workflow repositories. Changing parameter names / types too often will result in quite some work in updating / versioning the public workflows.

**Comment by hendrikweisser** [comment]
> I am just imaging that the tool FalseDiscoveryRate might be used to calculate FDRs in a different way (target decoy based is only one way to do so i.e. PEP based is also possible ). An explicit string might allow to extend it without breaking workflows again.

I don't think that should be the same parameter (if it ever gets implemented). Whether FDRs or q-values are reported is independent of which method is used to calculate them.

> I got bit more cautious when working with the KNIME guys and workflow repositories. Changing parameter names / types too often will result in quite some work in updating / versioning the public workflows.

Okay, but this pull request already contains changes to numerous parameters of different tools, so updates need to happen anyway. Cleaning up the "q_value" flag thus comes for free.

**Comment by timosachsenberg** [comment]
Adding a parameter (i.e. "prefix" or changing the description) is not breaking a KNIME workflow. Removing/renaming breaks them.
Just wanted to note that we should really consider the cost workflow updates we cause for our users.

**Comment by hendrikweisser** [comment]
From the change description:
> The "decoy_string" and "annotate_proteins" parameters were removed from FalseDiscoveryRate and PeptideIndexer, respectively.

So there are potentially workflow-breaking parameter changes independent of the "q_value" one already.

**Comment by timosachsenberg** [comment]
Ok. Thanks for fixing these issues.
